packages/aws-cdk-lib/aws-s3-deployment/lib/source.ts

```diff
@@ -262,5 +258,33 @@ export class Source {
     };
   }
 
+  /**
+   * Process objects such that it escapes token output suitable for JSON output.
+   *
+   * @param scope Parent construct scope
+   * @returns Object with with tokens escaped for JSON output.
+   */
+  private static escapeTokens(scope: Construct, obj: any): any {
+    if (Token.isUnresolved(obj)) {
+      // Return tokens as numbers. This is a hack to prevent the JSON serializer to wrap this token as a string.
```

Comment:
Is there a better approach which removes the need for the hack entirely?

Reply:
Only other approach I can think of is either:
- Create a custom JSON serializer
- Process the output of the JSON serialization such that the markers don't have string (`"..."`) escape.

Both methods are quite error-prone.

Reply:
> Create a custom JSON serializer
Probably nice, but what you have should work.